content/blog/what-is-an-algorithm/index.md

```diff
@@ -0,0 +1,212 @@
+
+## Operators
+
+In any programming language, you'll find a set of different symbols used to execute specific operations on either one variable, or between 2 or more variables.
+
+The simplest (and most common) ones are arithmetic operators: **Addition (+), Subtraction (-), Multiplication(x) and Division (÷)**
+
+> Note that in several programming languages, the 4 arithmetic operators can be used for more than just mathematical operations between numbers. A simple example is adding two words together in JavaScript to form another word, as such: `"Mathe" + "matical"`
+
+There are of course other operators that you'll get into as you start coding, but we won't get into all of them. Instead we'll look into one that is very common, and that's the **Equal sign =**
+As opposed to how it's usually used in simple arithmetics, as the part to indicate "this is the result", in most programming languages, it actually means something completely different.
+
+Let's take a look at the same example we used before, where we added two numbers together, and take the first instruction:
+
+`- Take the number 1`
+
+If we take into consideration the part where we talked about "storing values", we mentioned "**assigning**" values to variables. The **Equal sign (=)** is what we can use to _'put'_ values into variables. So a more exact way of describing that instruction would be:
+
+`- Assign the number 1 to a variable`
+
+With all of that put into consideration, let's try to write an algorithm that is a little bit more comprehensible to a computer:
+
+```elixir
+START
+	- Assign the number 1 to a variable
+	- Assign the number 2 to another variable
+	- Add the values of the two variables together
+	- Obtain the number 3
+END
+```
+
+All of this sounds great! However, we're still dealing with very simple operations. So what about big complex instructions? Like if we take a calculator, and try to calculate the **square root** of a certain number, how does it do that? Especially considering that the computer can't understand "Square root", and can only do very basic operations.
+
+To do that, we'd have to create specific "sets of instructions" that we can call for specific steps. Those sets of instructions are called "**Functions**"
+
+## Functions
+
+A function is a set of instructions that is defined _once_, but can be used _as many times as necessary_. In other words, it's a "pre-made chunk of simple instructions used to perform a complex task on demand".
+For different purposes, we can create different functions to make performing tasks easier for ourselves.
+
+The big advantage of making a function is to be able to quickly and easily use it without having to write the entire set of instructions over and over again, because over time you'll see that some tasks have to be performed hundreds of times in an applications, and to make all of that easier, functions are the way to go.
+
```

Comment:
Let's add another example in here using psuedo-code. Something like:

````suggestion

These instructions can come in handy, for example, when trying to do bulk-tasks, like cooking multiple pies at once.

```elixir
BAKE_START
   - Pre-heat the oven to 350
   - Put the pie in the oven
   - Wait an hour
   - Take the pie out of the oven
BAKE_END

START
	- Prepare Apple Pie
	- Prepare Rhubarb Pie
	- BAKE Apple Pie
	- BAKE Rhubarb Pie 
END
```

````

Reply:
Great suggestion! I added it and changed the wording a bit for more clarity.